### PR TITLE
Adding support to generate ARM Templates

### PR DESCRIPTION
**Pull request by uday31in**
Adding support to generate ARM Template

### CODE REVIEW

**Review by krnese — APPROVED** [review]

File: src/data/template/Microsoft.Authorization/policyDefinitions.template.jq [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {},
+    "variables": {},
+    "resources": [
+        {
+            "type": .ResourceType,
+            "name": .Name,
+            "apiVersion": "0000-00-00",
+            "location": "[deployment().location]",
```
Comment (krnese) [inline src/data/template/Microsoft.Authorization/policyDefinitions.template.jq:11]: location should be removed

File: src/data/template/Microsoft.Authorization/policySetDefinitions.template.jq [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {},
+    "variables": {},
+    "resources": [
+        {
+            "type": .ResourceType,
+            "name": .Name,
+            "apiVersion": "0000-00-00",
+            "location": "[deployment().location]",
```
Comment (krnese) [inline src/data/template/Microsoft.Authorization/policySetDefinitions.template.jq:11]: location should be removed

File: src/data/template/Microsoft.Authorization/roleAssignments.template.jq [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {},
+    "variables": {},
+    "resources": [
+        {
+            "type": .ResourceType,
+            "name": .Name,
+            "apiVersion": "0000-00-00",
+            "location": "[deployment().location]",
```
Comment (krnese) [inline src/data/template/Microsoft.Authorization/roleAssignments.template.jq:11]: location can be removed

File: src/data/template/Microsoft.Authorization/roleDefinitions.template.jq [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {},
+    "variables": {},
+    "resources": [
+        {
+            "type": .ResourceType,
+            "name": .Name,
+            "apiVersion": "0000-00-00",
+            "location": "[deployment().location]",
```
Comment (krnese) [inline src/data/template/Microsoft.Authorization/roleDefinitions.template.jq:11]: same

File: src/data/template/Microsoft.Authorization/policyAssignments.template.jq [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {},
+    "variables": {},
+    "resources": [
+        {
+            "type": .ResourceType,
+            "name": .Name,
+            "apiVersion": "0000-00-00",
+            "location": "[deployment().location]",
```
Comment (krnese) [inline src/data/template/Microsoft.Authorization/policyAssignments.template.jq:11]: location is only needed when system MI is required for DINE/Modify

**Review by krnese — APPROVED** [review]

File: src/data/template/Microsoft.Authorization/policyAssignments.jq [hunk 6]
```diff
@@ -0,0 +1 @@
+del(.ResourceId, .Properties.Metadata, .ResourceGroupName, .SubscriptionId)
```
Comment (krowlandson) [inline src/data/template/Microsoft.Authorization/policyAssignments.jq:1]: Appreciate this probably works, but shouldn't these keys be in camelCase?

Comment (uday31in) [reply src/data/template/Microsoft.Authorization/policyAssignments.jq:1]: JQ is case sensitive and Built-in PowerShell class has these names.

File: src/data/template/Microsoft.Authorization/policyAssignments.template.jq (hunk 5 above)
Comment (uday31in) [reply src/data/template/Microsoft.Authorization/policyAssignments.template.jq:11]: can we keep it then as we dont know whether it is is DINE policy assignment or not?

Comment (krnese) [reply src/data/template/Microsoft.Authorization/policyAssignments.template.jq:11]: Yes

**Review by krnese — APPROVED** [review]

File: src/data/template/Microsoft.Authorization/policyAssignments.jq (hunk 6 above)
Comment (uday31in) [reply src/data/template/Microsoft.Authorization/policyAssignments.jq:1]: image

**Pull request merged**